src/cmd-diff

```diff
@@ -221,31 +221,50 @@ def ensure_extracted_rootfses(diff_from, diff_to):
     return (dir_from, dir_to)
 
 
-def diff_live_rootfs(diff_from, diff_to):
+def diff_live_rootfs_img(diff_from, diff_to):
     (dir_from, dir_to) = ensure_extracted_rootfses(diff_from, diff_to)
     git_diff(dir_from, dir_to)
 
 
-def diff_live_squashfs_tree(diff_from, diff_to):
-    (dir_from, dir_to) = ensure_extracted_rootfses(diff_from, diff_to)
-    diff_cmd_outputs(['unsquashfs', '-d', '', '-l', '-excludes', '{}',
-                      '/ostree/deploy', '/ostree/repo/objects'],
-                     os.path.join(dir_from, "root.squashfs"),
-                     os.path.join(dir_to, "root.squashfs"))
+def extract_live_sysroot_img(diff_from, diff_to):
+    (rootfs_dir_from, rootfs_dir_to) = ensure_extracted_rootfses(diff_from, diff_to)
 
+    def get_sysroot_img(dir):
+        for fstype in ["erofs", "squashfs"]:
+            img = os.path.join(dir, f"root.{fstype}")
+            if os.path.exists(img):
+                return img
+        raise Exception(f"Cannot find root.[squash|ero]fs in {dir}")
 
-def diff_live_squashfs(diff_from, diff_to):
-    (rootfs_dir_from, rootfs_dir_to) = ensure_extracted_rootfses(diff_from, diff_to)
-    squashfs_from = os.path.join(rootfs_dir_from, "root.squashfs")
-    squashfs_to = os.path.join(rootfs_dir_to, "root.squashfs")
-    dir_from = os.path.join(cache_dir("live-squashfs"), diff_from.id)
-    dir_to = os.path.join(cache_dir("live-squashfs"), diff_to.id)
+    def run_unpack_sysroot_img(img, dest):
+        if img.endswith(".squashfs"):
+            runcmd(['unsquashfs', '-d', dest, '-no-xattrs', '-excludes', img, '/ostree/deploy', '/ostree/repo/objects'])
+        else:
+            runcmd(['fsck.erofs', f'--extract={dest}', '--no-xattrs', img])
+            runcmd(['sudo', 'rm', '-r', '-f', os.path.join(dest, "ostree/deploy")], quiet=True)
+            runcmd(['sudo', 'rm', '-r', '-f', os.path.join(dest, "ostree/repo/objects")], quiet=True)
+
+    fs_from = get_sysroot_img(rootfs_dir_from)
+    fs_to = get_sysroot_img(rootfs_dir_to)
+
+    dir_from = os.path.join(cache_dir("live-sysroot-img"), diff_from.id)
+    dir_to = os.path.join(cache_dir("live-sysroot-img"), diff_to.id)
 
     if not os.path.exists(dir_from):
-        runcmd(['unsquashfs', '-d', dir_from, '-no-xattrs', '-excludes', squashfs_from, '/ostree/deploy', '/ostree/repo/objects'])
+        run_unpack_sysroot_img(fs_from, dir_from)
     if not os.path.exists(dir_to):
-        runcmd(['unsquashfs', '-d', dir_to, '-no-xattrs', '-excludes', squashfs_to, '/ostree/deploy', '/ostree/repo/objects'])
+        run_unpack_sysroot_img(fs_to, dir_to)
+
+    return (dir_from, dir_to)
+
+
+def diff_live_sysroot_tree(diff_from, diff_to):
+    (dir_from, dir_to) = extract_live_sysroot_img(diff_from, diff_to)
+    diff_cmd_outputs(['find', '{}', '-printf', "%P\n"], dir_from, dir_to)
```

Comment:
Hmm right, that's much more expensive now that we have to extract the full squashfs/erofs but there isn't really a listing equivalent for erofs we could use here (there's `dump.erofs --ls`, but there's no option to make it recurse) which I guess is how you ended up here. And I guess it's nice that it allows comparing across the squashfs -> erofs transition.

Reply:
That said, I think we want to sort the output of `find` since it doesn't guarantee lexicographic order which is important here for a good diff. We could add an option to `diff_cmd_outputs` to pipe the output of the cmd to `sort`.

Reply:
>there's no option to make it recurse) which I guess is how you ended up here

Yes, and anyhow we unpack it in other `diff` cmd, so it's expensive, but probably we can stay with it. Other option - there are some tools available for `erofs`, but i haven't tested them